overlays/nodejs/lambda/lambdaMixins.ts

```diff
@@ -0,0 +1,365 @@
+
+    /**
+     * Information about the Amazon Cognito identity provider when invoked through the AWS Mobile
+     * SDK. It can be null.
+     */
+    readonly identity: any;
+
+    /**
+     * Information about the client application and device when invoked through the AWS Mobile SDK.
+     * It can be null.
+     */
+    readonly clientContext: any;
+
+    /**
+     * Returns the approximate remaining execution time (before timeout occurs) of the Lambda
+     * function that is currently executing. The timeout is one of the Lambda function
+     * configuration. When the timeout reaches, AWS Lambda terminates your Lambda function.
+     *
+     * You can use this method to check the remaining time during your function execution and take
+     * appropriate corrective action at run time.
+     */
+    getRemainingTimeInMillis(): string;
+}
+
+/**
+ * EntryPoint is the signature for a serverless function that will be invoked each time the AWS
+ * Lambda is invoked.
+ *
+ * This function can be synchronous or asynchronous function, though async is only supported with an
+ * AWS Lambda runtime of 8.10 or higher.  On those runtimes a Promise can be returned, 'callback'
+ * parameter can be ignored, and AWS will appropriately handle things. For AWS lambda pre-8.10, a
+ * synchronous function must be provided.  The synchronous function should return nothing, and
+ * should instead invoke 'callback' when complete.
+ */
+export type EntryPoint<E, R> = (event: E, context: Context, callback: (error: any, result: R) => void) => Promise<R> | void;
+
+/**
+ * EntryPointFactory is the signature for a function that will be called once to produce the
+ * serverless function that AWS Lambda will invoke.  It can be used to initialize expensive state
+ * once that can then be used across all invocations of the Lambda (as long as the Lambda is using
+ * the same warm node instance).
+ */
+export type EntryPointFactory<E, R> = () => EntryPoint<E, R>;
+
+/**
+ * An EventHandler is either a JavaScript Function instance or an aws.lambda.Function that can be
+ * used to handle an event triggered by some resource.  If just a JavaScript function is provided
+ * the AWS Lambda will be created by calling [createFunction] on it.  If more control over the
+ * resultant AWS Lambda is required, clients can call [createFunction] directly and pass the result
+ * of that to any code that needs an EventHandler.
+ */
+export type EventHandler<E, R> = EntryPoint<E, R> | LambdaFunction;
+
+/**
+ * FunctionOptions provides configuration options for the serverless Function.  It is effectively
+ * equivalent to [aws.lambda.FunctionArgs] except with a few important differences documented at the
+ * property level.  For example, [role] is an actual iam.Role instance, and not an ARN. Properties
+ * like [runtime] are now optional.  And some properties (like [code]) are entirely disallowed.
+ */
+export type FunctionOptions<E, R> = utils.Overwrite<FunctionArgs, {
```

Comment:
It seems a little strange now that this module has both `aws.lambda.FunctionArgs` and `aws.lambda.FunctionOptions`.  Should the latter be `aws.lambda.CallbackFunctionArgs`?

Reply:
Ok.  Based on this the above feedack, i'm going to be naming these to FunctionCallback, FunctionCallbackFactory, and FunctionCallbackArgs